From a44ab56c8617af31c55436657fdc47504a1e8efc Mon Sep 17 00:00:00 2001
From: David Wronek <david-w@ctemplar.com>
Date: Wed, 17 Nov 2021 14:59:35 +0100
Subject: [PATCH 201/210] arm64: dts: qcom: msm8916-huawei-y635: Add display

Signed-off-by: David Wronek <david-w@ctemplar.com>
---
 .../boot/dts/qcom/msm8916-huawei-y635.dts     | 50 +++++++++++++++++++
 1 file changed, 50 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
index 5c7a2f90cf1..e8ab0fc3ec7 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
@@ -100,6 +100,39 @@ &blsp1_uart2 {
 	status = "okay";
 };
 
+&dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		compatible = "huawei,boe-otm8019a";
+		reg = <0>;
+
+		power-supply = <&pm8916_l17>;
+		reset-gpios = <&msmgpio 25 GPIO_ACTIVE_LOW>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi0_out>;
+			};
+		};
+	};
+};
+
+&dsi0_out {
+	data-lanes = <1 2>;
+	remote-endpoint = <&panel_in>;
+};
+
+&dsi_phy0 {
+	qcom,dsi-phy-regulator-ldo-mode;
+};
+
+&mdss {
+	status = "okay";
+};
+
 &pm8916_resin {
 	status = "okay";
 	linux,code = <KEY_VOLUMEDOWN>;
@@ -265,6 +298,23 @@ gpio_keys_default: gpio-keys-default {
 		bias-pull-up;
 	};
 
+	mdss {
+		mdss_default: mdss-default {
+			pins = "gpio25";
+			function = "gpio";
+
+			drive-strength = <8>;
+			bias-disable;
+		};
+		mdss_sleep: mdss-sleep {
+			pins = "gpio25";
+			function = "gpio";
+
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
+
 	prox_irq_default: prox-irq-default {
 		pins = "gpio113";
 		function = "gpio";
-- 
2.34.1

