diff --git a/arch/arm64/boot/dts/amlogic/Makefile b/arch/arm64/boot/dts/amlogic/Makefile
index ced039463..58e79f81e 100644
--- a/arch/arm64/boot/dts/amlogic/Makefile
+++ b/arch/arm64/boot/dts/amlogic/Makefile
@@ -1,4 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0
+dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905l-r3300l.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905l2-e900v21e.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905l3-b860av21u.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-g12a-s905l3a-e900v22d.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-a1-ad401.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-axg-jethome-jethub-j100.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-axg-s400.dtb
diff --git a/arch/arm64/boot/dts/amlogic/meson-g12a-s905l3a-e900v22d.dts b/arch/arm64/boot/dts/amlogic/meson-g12a-s905l3a-e900v22d.dts
new file mode 100644
index 000000000..c5d50cff7
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-g12a-s905l3a-e900v22d.dts
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2018 Amlogic, Inc. All rights reserved.
+ * Create by flippy
+ */
+
+/dts-v1/;
+
+#include "meson-g12a-u200.dts"
+
+/ {
+	compatible = "skyworth,e900v22d", "amlogic,g12a";
+	model = "SKYWORTH E900V22D";
+
+	vddgpu: regulator-vddgpu {
+		compatible = "regulator-fixed";
+		regulator-name = "mali";
+		regulator-min-microvolt = <800000>;
+		regulator-max-microvolt = <800000>;
+		vin-supply = <&main_12v>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	wifi32k: wifi32k {
+		compatible = "pwm-clock";
+		#clock-cells = <0>;
+		clock-frequency = <32768>;
+		pwms = <&pwm_ef 0 30518 0>; /* PWM_E at 32.768KHz */
+	};
+
+	sdio_pwrseq: sdio-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&gpio GPIOX_6 GPIO_ACTIVE_LOW>;
+		clocks = <&wifi32k>;
+		clock-names = "ext_clock";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		
+		power_led {
+			led_name = "power_led";
+                        gpios = <&gpio_ao GPIOAO_11 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		net_led {
+			led_name = "net_led";
+                        gpios = <&gpio_ao GPIOAO_9 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "0.0:00:link";
+		};
+
+		remote_led {
+			led_name = "remote_led";
+                        gpios = <&gpio_ao GPIOAO_10 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+			linux,default-trigger = "rc-feedback";
+		};
+
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+};
+
+&mali {
+	mali-supply=<&vddgpu>;
+};
+
+/* SDIO */
+&sd_emmc_a {
+	status = "okay";
+	pinctrl-0 = <&sdio_pins>;
+	pinctrl-1 = <&sdio_clk_gate_pins>;
+	pinctrl-names = "default", "clk-gate";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	bus-width = <4>;
+	cap-sd-highspeed;
+	sd-uhs-sdr50;
+	max-frequency = <100000000>;
+
+	non-removable;
+	disable-wp;
+
+	/* WiFi firmware requires power to be kept while in suspend */
+	keep-power-in-suspend;
+
+	mmc-pwrseq = <&sdio_pwrseq>;
+
+	vmmc-supply = <&vddao_3v3>;
+	vqmmc-supply = <&vddao_1v8>;
+
+	wifi: wifi@1 {
+		reg = <1>;
+		compatible = "sprd,unisoc-wifi";
+	};
+};
+
+/* SD card */
+&sd_emmc_b {
+	status = "okay";
+	bus-width = <4>;
+	cap-sd-highspeed;
+	max-frequency = <50000000>;
+};
+
+/* eMMC */
+&sd_emmc_c {
+	status = "okay";
+	bus-width = <8>;
+	cap-mmc-highspeed;
+	mmc-ddr-1_8v;
+	mmc-hs200-1_8v;
+	max-frequency = <50000000>;
+};
+
+/* This is connected to the Bluetooth module: */
+&uart_A {
+	status = "okay";
+	/* pinctrl-0 = <0x55 0x56>; #指向uart-a-pins和uart-a-cts-rts-pins */
+	pinctrl-0 = <&uart_a_pins>, <&uart_a_cts_rts_pins>;
+	pinctrl-names = "default";
+	uart-has-rtscts;
+};
+
+/* CPU Overclock */
+&cpu_opp_table {
+	opp-100000000 {
+		status = "disabled";
+	};
+
+	opp-250000000 {
+		status = "disabled";
+	};
+
+	opp-500000000 {
+		status = "disabled";
+	};
+
+	opp-667000000 {
+		status = "disabled";
+	};
+
+	opp-1908000000 {
+		opp-hz = /bits/ 64 <1908000000>;
+		opp-microvolt = <1001000>;
+	};
+
+	/* 2016mhz :  not test */
+	//opp-2016000000 {
+	//		opp-hz = /bits/ 64 <2016000000>;
+	//		opp-microvolt = <1021000>;
+	//};
+};
diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905l-r3300l.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905l-r3300l.dts
new file mode 100644
index 000000000..90706d4ae
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905l-r3300l.dts
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * hf
+ */
+
+/dts-v1/;
+
+#include "meson-gxl-s905x-p212.dts"
+
+/ {
+	compatible = "bestv,r3300l", "amlogic,s905l", "amlogic,meson-gxl";
+	model = "BESTV R3300L";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x40000000>;
+	};
+};
+
+/* Wireless SDIO Module */
+&sd_emmc_a {
+	status = "okay";
+	/*
+	 * Explicitly define the sdio device, so that we can add an ethernet
+	 * alias for it (which e.g. makes u-boot set a mac-address).
+	 */
+	rtl8189ftv: sdio_wifi@1 {
+		reg = <1>;
+	};
+};
+
+/* eMMC */
+&sd_emmc_c {
+	status = "okay";
+};
+
+/* This is connected to the Bluetooth module: */
+&uart_A {
+	status = "okay";
+	/delete-node/ bluetooth;
+};
\ No newline at end of file
diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905l2-e900v21e.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905l2-e900v21e.dts
new file mode 100644
index 000000000..e8458afb5
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905l2-e900v21e.dts
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2016 Endless Computers, Inc.
+ * Author: Carlo Caione <carlo@endlessm.com>
+ */
+
+/dts-v1/;
+
+#include "meson-gxl-s905w-p281.dts"
+
+/ {
+	compatible = "skyworth,e900v21e", "amlogic,s905l2", "amlogic,meson-gxl";
+	model = "SKYWORTH E900V21E";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x40000000>;
+	};
+};
+
+// the gpu in the S905L2 has one pp core less
+// (i.e. only two in total) than the one in the other S905X/W
+&mali {
+	status="disabled";
+	interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 164 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 165 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 166 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 167 IRQ_TYPE_LEVEL_HIGH>;
+	interrupt-names = "gp", "gpmmu", "pp", "pmu",
+		"pp0", "ppmmu0", "pp1", "ppmmu1";
+};
+
+&hdmi_tx {
+	status = "disabled";
+};
+
+/* eMMC */
+&sd_emmc_c {
+	status = "okay";
+	max-frequency = <25000000>;
+};
\ No newline at end of file
diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905l3-b860av21u.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905l3-b860av21u.dts
new file mode 100644
index 000000000..b014a5446
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905l3-b860av21u.dts
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2016 Endless Computers, Inc.
+ * Author: Carlo Caione <carlo@endlessm.com>
+ */
+
+/dts-v1/;
+
+#include "meson-gxl-s905w-p281.dts"
+
+/ {
+	compatible = "zte,b860av21u", "amlogic,s905l3", "amlogic,meson-gxl";
+	model = "ZTE B860AV21U";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x40000000>;
+	};
+};
+
+/* GPU */
+// the gpu in the S905L2 has one pp core less
+// (i.e. only two in total) than the one in the other S905X/W
+&mali {
+	status="disabled";
+	interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 164 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 165 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 166 IRQ_TYPE_LEVEL_HIGH>,
+		<GIC_SPI 167 IRQ_TYPE_LEVEL_HIGH>;
+	interrupt-names = "gp", "gpmmu", "pp", "pmu",
+		"pp0", "ppmmu0", "pp1", "ppmmu1";
+};
+
+&hdmi_tx {
+	status = "disabled";
+};
+
+/* eMMC */
+&sd_emmc_c {
+	status = "okay";
+	max-frequency = <50000000>;
+};
